controllers/application_poller_controller.go

```diff
@@ -0,0 +1,288 @@
+		p.handleErrors(ctx, fmt.Errorf("%s (%s): %w", "failed to assemble application", activity.URL, err), activity.URL)
+		return
+	}
+
+	generatedResources, err := p.generateApp(*assembledApp)
+	if err != nil {
+		p.handleErrors(ctx, fmt.Errorf("%s (%s): %w", "failed to generate application", activity.URL, err), activity.URL)
+		return
+	}
+
+	var exp *optimizev1beta2.Experiment
+	for i := range generatedResources {
+		if expObj, ok := generatedResources[i].(*optimizev1beta2.Experiment); ok {
+			exp = expObj
+			break
+		}
+	}
+
+	if exp == nil {
+		p.handleErrors(ctx, fmt.Errorf("%s: %w", "invalid experiment generated", err), activity.URL)
+		return
+	}
+
+	switch activity.Tags[0] {
+	case applications.TagScan:
+		template, err := server.ClusterExperimentToAPITemplate(exp)
+		if err != nil {
+			p.handleErrors(ctx, fmt.Errorf("%s: %w", "failed to convert experiment template", err), activity.URL)
+			return
+		}
+
+		if err := p.apiClient.UpdateTemplate(ctx, templateURL, *template); err != nil {
+			p.handleErrors(ctx, fmt.Errorf("%s: %w", "failed to save experiment template in server", err), activity.URL)
+			return
+		}
+
+		p.log.Info("successfully completed resource scan", "activity", activity.URL)
+	case applications.TagRun:
+		// We wont compare existing scan with current scan
+		// so we can preserve changes via UI
+
+		// Get previous template
+		previousTemplate, err := p.apiClient.GetTemplate(ctx, templateURL)
+		if err != nil {
+			p.handleErrors(ctx, fmt.Errorf("%s: %w", "failed to get experiment template from server, a 'scan' task must be completed first", err), activity.URL)
+			return
+		}
+
+		// Overwrite current scan results with previous scan results
+		if err = server.APITemplateToClusterExperiment(exp, &previousTemplate); err != nil {
+			p.handleErrors(ctx, fmt.Errorf("%s: %w", "failed to convert experiment template", err), activity.URL)
+			return
+		}
+
+		// At this point the experiment should be good to create/deploy/run
+		// so let's create all the resources and #profit
+
+		// TODO
+		// try to clean up on failure ( might be a simple / blind p.client.Delete(ctx,generatedResources[i])
+		for i := range generatedResources {
+			// TODO generatedResource ( experiment ) does not contain the namespace
+			// not sure why yet
+			objKey, err := client.ObjectKeyFromObject(generatedResources[i])
+			if err != nil {
+				p.handleErrors(ctx, fmt.Errorf("%s: %w", "failed to get object key", err), activity.URL)
+				return
+			}
+
+			holder := &unstructured.Unstructured{}
+			holder.SetGroupVersionKind(generatedResources[i].GetObjectKind().GroupVersionKind())
+			err = p.client.Get(ctx, objKey, holder)
+			switch {
+			case apierrors.IsNotFound(err):
+				if err := p.client.Create(ctx, generatedResources[i]); err != nil {
+					p.handleErrors(ctx, fmt.Errorf("%s: %w", "failed to create object", err), activity.URL)
+					return
+				}
+			case err == nil:
```

Comment:
lets not support updates for this round; 
log a message saying this would have triggered an update, but don't do anything